src/getBatchPosters.ts

```diff
@@ -0,0 +1,181 @@
+  const [batchPoster, isAdd] = getBatchPostersFromFunctionData({
+    abi: [setIsBatchPosterABI],
+    data: input,
+  });
+
+  if (isAdd) {
+    acc.add(batchPoster);
+  } else {
+    acc.delete(batchPoster);
+  }
+
+  return acc;
+}
+
+export type GetBatchPostersParams = {
+  /** Address of the rollup we're getting list of batch posters from */
+  rollup: Address;
+  /** Address of the sequencerInbox we're getting logs from */
+  sequencerInbox: Address;
+};
+export type GetBatchPostersReturnType = {
+  /**
+   * If logs contain unknown signature, batch posters list might:
+   * - contain false positives (batch posters that were removed, but returned as batch poster)
+   * - contain false negatives (batch posters that were added, but not present in the list)
+   */
+  isAccurate: boolean;
+  /** List of batch posters for the given rollup */
+  batchPosters: Address[];
+};
+
+/**
+ *
+ * @param {PublicClient} publicClient - The chain Viem Public Client
+ * @param {GetBatchPostersParams} GetBatchPostersParams {@link GetBatchPostersParams}
+ *
+ * @returns Promise<{@link GetBatchPostersReturnType}>
+ *
+ * @remarks Batch posters list is not guaranteed to be exhaustive if the `isAccurate` flag is false.
+ * It might contain false positive (batch posters that were removed, but returned as batch poster)
+ * or false negative (batch posters that were added, but not present in the list)
+ *
+ * @example
+ * const { isAccurate, batchPosters } = getBatchPosters(client, {
+ *   rollup: '0xc47dacfbaa80bd9d8112f4e8069482c2a3221336',
+ *   sequencerInbox: '0x995a9d3ca121D48d21087eDE20bc8acb2398c8B1'
+ * });
+ *
+ * if (isAccurate) {
+ *   // batch posters were all fetched properly
+ * } else {
+ *   // batch posters list is not guaranteed to be accurate
+ * }
+ */
+export async function getBatchPosters<TChain extends Chain | undefined>(
+  publicClient: PublicClient<Transport, TChain>,
+  { rollup, sequencerInbox }: GetBatchPostersParams,
+): Promise<GetBatchPostersReturnType> {
+  let blockNumber: bigint | 'earliest';
+  let createRollupTransactionHash = null;
+  try {
+    createRollupTransactionHash = await createRollupFetchTransactionHash({
+      rollup,
+      publicClient,
+    });
+    const receipt = await publicClient.waitForTransactionReceipt({
+      hash: createRollupTransactionHash,
+    });
+    blockNumber = receipt.blockNumber;
+  } catch (e) {
+    blockNumber = 'earliest';
+  }
+
+  const sequencerInboxEvents = await publicClient.getLogs({
+    address: sequencerInbox,
+    event: ownerFunctionCalledEventAbi,
+    args: { id: 6n },
+    fromBlock: blockNumber,
+    toBlock: blockNumber,
+  });
+
+  const events = createRollupTransactionHash
+    ? [{ transactionHash: createRollupTransactionHash }, ...sequencerInboxEvents]
+    : sequencerInboxEvents;
+  const txs = await Promise.all(
+    events.map((event) =>
+      publicClient.getTransaction({
+        hash: event.transactionHash,
+      }),
+    ),
+  );
```

Comment:
This is the main difference, we need to get `createRollup` event data from `rollup` contract, but rest of the logs from `sequencerInbox`